commit d31211b9d47360f044fc63e67deaec3605f8d081 (tag: v1.2.1, refs/changes/28/2628/2)
Author: David Pacheco <dap@joyent.com>
Date:   2017-09-20T14:53:21-07:00 (2 years, 1 month ago)
    
    joyent/mdb_v8#96 jsclosure crash on invalid context index
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 2106196..933871c 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -14,6 +14,10 @@
 
 None.
 
+## v1.2.1 (2017-09-20)
+
+* #96 jsclosure crash on invalid context index
+
 ## v1.2.0 (2017-08-31)
 
 * #44 want "jsfunction" dcmd
diff --git a/README.md b/README.md
index 27642a9..afb1c34 100644
--- a/README.md
+++ b/README.md
@@ -44,17 +44,17 @@ https://us-east.manta.joyent.com/Joyent_Dev/public/mdb_v8.  If you have the
 can list the latest binaries with:
 
     $ mfind -t o $(mget -q /Joyent_Dev/public/mdb_v8/latest)
-    /Joyent_Dev/public/mdb_v8/v1.2.0/mdb_v8_amd64.so
-    /Joyent_Dev/public/mdb_v8/v1.2.0/mdb_v8_ia32.so
+    /Joyent_Dev/public/mdb_v8/v1.2.1/mdb_v8_amd64.so
+    /Joyent_Dev/public/mdb_v8/v1.2.1/mdb_v8_ia32.so
 
 You can fetch a specific binary like this (in this case, the 32-bit version
-1.2.0 binary):
+1.2.1 binary):
 
-    $ mget -O /Joyent_Dev/public/mdb_v8/v1.2.0/mdb_v8_ia32.so
+    $ mget -O /Joyent_Dev/public/mdb_v8/v1.2.1/mdb_v8_ia32.so
 
 or using curl:
 
-    $ curl -O https://us-east.manta.joyent.com/Joyent_Dev/public/mdb_v8/v1.2.0/mdb_v8_ia32.so
+    $ curl -O https://us-east.manta.joyent.com/Joyent_Dev/public/mdb_v8/v1.2.1/mdb_v8_ia32.so
 
 This one-liner will get you the latest 32-bit binary:
 
diff --git a/src/mdb_v8_function.c b/src/mdb_v8_function.c
index a2b1687..05276a8 100644
--- a/src/mdb_v8_function.c
+++ b/src/mdb_v8_function.c
@@ -677,7 +677,7 @@ v8context_var_value(v8context_t *ctxp, unsigned int i, uintptr_t *valptr)
 	unsigned int idx;
 
 	idx = i + V8_CONTEXT_NCOMMON;
-	if (i >= ctxp->v8ctx_nelts) {
+	if (idx >= ctxp->v8ctx_nelts) {
 		v8_warn("context %p: variable index %d is out of range\n",
 		    ctxp->v8ctx_addr, i);
 		return (-1);
diff --git a/version b/version
index 9a05c1a..0159247 100644
--- a/version
+++ b/version
@@ -8,4 +8,4 @@
 #
 # This value is also used by the "publish" target in the build process.
 #
-1.2.0
+1.2.1
